phac-nml/irida-next
# Email notifications for user access through group links

## What does this PR do and why?
Implemented emails for notifying users and managers when a group has been invited/uninvited to/from a group/project.
Closes #476.

Note: A manager is a member that has a role of owner or maintainer.

## Screenshots or screen recordings
Access **granted** email being sent to a **user**:
![image](https://github.com/phac-nml/irida-next/assets/325703/7ff27e59-db27-4c5c-a8e3-6336fe405759)
Access **granted** email being sent to a **manager**:
![image](https://github.com/phac-nml/irida-next/assets/325703/ed0e5e4d-b409-4a66-98cd-0748215f7fc4)
Access **revoked** email being sent to a **user**:
![image](https://github.com/phac-nml/irida-next/assets/325703/a4ba12ab-812d-44d9-9162-d5335476cd7c)
Access **revoked** email being sent to a **manager**:
![image](https://github.com/phac-nml/irida-next/assets/325703/87646141-f995-4145-98cb-dab645b3723b)

## How to set up and validate locally
1. Install local mock email service, [Mailhog](https://github.com/mailhog/MailHog).
1. After setup, open a terminal and run `~/go/bin/MailHog`.
1. Run irida-next.
1. Login to irida-next.
1. Navigate to a `group` or `project` that the logged in user is a manager of. Alternatively, you could create a new `group` or `project`.
1. Select the `Members` menu item, then `Groups` tab.
1. Try inviting & uninviting a group.
1. Open a browser and navigate to http://0.0.0.0:8025/. Check the mock email service for sent emails. 

Note: Email previews are located at http://localhost:3000/rails/mailers/group_link_mailer.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
